src/OAINet.Node/Network/Node.cs

```diff
@@ -51,12 +60,25 @@ private void RegisterHandlers()
     }
     public async Task RunNode()
     {
-        _peer = InitializeNode(3024);
+        _peer = InitializeNode();
+        if (_peer == null)
+        {
+            _logger.LogInformation("the node cannot start.");
+            return;
+        }
+
+        NodePeerChecker();
         await AcceptRequest();
         Console.ReadKey();
```

Comment:
Is it possible to use dependency injection ?

Reply:
no.